test/ApplicationTest.php

```diff
@@ -128,6 +129,20 @@ public function testPipeCanAcceptAPathArgument($middleware)
         $this->assertNull($this->app->pipe('/foo', $middleware));
     }
 
+    public function testPipeNonSlashPathOnNonStringPipeProduceTypeError()
+    {
+        // @codingStandardsIgnoreStart
+        $middleware1 = function ($request, $response) {};
+        // @codingStandardsIgnoreEnd
+        $middleware2 = $this->createMockMiddleware();
+
+        try {
+            $this->app->pipe($middleware1, $middleware2);
+        } catch (TypeError $t) {
```

Comment:
Can't we use `$this->expectedException`?

Reply:
updated.